src/librustc_middle/query/mod.rs

```diff
@@ -386,8 +386,15 @@ rustc_queries! {
             storage(ArenaCacheSelector<'tcx>)
         }
 
-        /// HACK: when evaluated, this reports a "unsafe derive on repr(packed)" error
-        query unsafe_derive_on_repr_packed(_: DefId) -> () {}
+        /// HACK: when evaluated, this reports a "unsafe derive on repr(packed)" error.
+        ///
+        /// Unsafety checking is executed for each method seperately, but we only want
+        /// to emit this error once per derive. As there are some impls with multiple
+        /// methods, we use a query for deduplication.
+        query unsafe_derive_on_repr_packed(key: LocalDefId) -> () {
+            desc { |tcx| "processing `{}`", tcx.def_path_str(key.to_def_id()) }
+            storage(ArenaCacheSelector<'tcx>)
```

Comment:
This change seems unrelated to this PR?

Reply:
Afaik we have to explicitly mention the storage and desc in case the key is not a `DefId`.

Reply:
From what I see, most of queries taking `LocalDefId` don't specify storage.